drv/sprot-api/src/lib.rs

```diff
@@ -71,173 +273,73 @@ pub enum SprotError {
     BadMessageType,
     /// Transfer size is outside of maximum and minimum lenghts for message type.
     BadMessageLength,
-    /// Error from Spi
-    SpiServerError,
-    /// Message is too large
-    Oversize,
-    /// Tx buffer is unexpectedly not Idle.
-    TxNotIdle,
-
-    // SPI-related errors
-    SpiServerLockError,
-    SpiServerWritePart1Error,
-    SpiServerWritePart2Error,
-    SpiServerReleaseError,
-    SpiServerReadPart1Error,
-    SpiServerReadPart2Error,
+    // We cannot assert chip select
     CannotAssertCSn,
-    RotNotReady,
-    // SP initiated RoT reset request should timeout.
-    RspTimeout,
-    BadResponse,
-    RotBusy,
-
-    /// Feature is not implemented
-    NotImplemented,
-    /// An error code reserved for the SP was used by the Rot
-    NonRotError,
-    /// A message with version = 0 was received unexpectedly.
-    EmptyMessage,
-
-    /// Insufficient bytes received
-    Incomplete,
-    /// Hubpack error
+    // The request timed out
+    Timeout,
+    // Hubpack error: Used for both serialization and deserialization
```

Comment:
(I _suspect_ you can eliminate this. I suspect all the hubpack error cases are programmer errors that can become `unpack_lite`. But I'm not 100% sure.)

Reply:
There can be deserialization errors on short reads, so I changed it to Deserialization.